src/ace/managers/state.c

```diff
@@ -0,0 +1,181 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <ace/managers/state.h>
+#include <ace/managers/log.h>
+
+/* Functions */
+
+#ifdef ACE_DEBUG
+void _checkPointers(const char *szPointerName, void *pPointer) {
+	if (!pPointer) {
+		logWrite("ERR: Pointer %s is zero! Crash is eminent!\n", szPointerName);
+	}
+}
+#endif
+
+tStateManager *stateManagerCreate(void) {
+	logBlockBegin("stateManagerCreate()");
+
+	tStateManager *pStateManager = memAllocFastClear(sizeof(tStateManager));
+
+	logBlockEnd("stateManagerCreate()");
+
+	return pStateManager;
+}
+
+void stateManagerDestroy(tStateManager *pStateManager) {
+	logBlockBegin("stateManagerDestroy(pStateManager: %p)", pStateManager);
+
+#ifdef ACE_DEBUG
```

Comment:
if you call fn via macro, the macro def can be empty in release builds, so no need for ifdefs butchering our code.